docs/theory/dft/functional_derivatives.md

```diff
@@ -0,0 +1,39 @@
+## Functional derivatives
+
+In the last section the functional derivative
+
+$$\hat F_{\rho_\alpha}^\mathrm{res}(r)=\left(\frac{\delta F^\mathrm{res}}{\delta\rho_\alpha(r)}\right)_{T,\rho_{\alpha'\neq\alpha}}$$
+
+was introduced as part of the Euler-Lagrange equation. The implementation of these functional derivatives can be a major difficulty during the development of a new Helmholtz energy model. In $\text{FeO}_\text{s}$ it is fully automated. The core assumption is that the residual Helmholtz energy functional $\hat F^\mathrm{res}$ can be written as a sum of contributions that each can be written in the following way:
+
+$$F=\int f[\rho(r)]dr=\int f(\lbrace n_\gamma\rbrace)dr$$
+
+The Helmholtz energy density $f$ which would in general be a functional of the density itself can be expressed as a function of weighted densities $n_\gamma$ which are obtained by convolving the density profiles with weight functions $\omega_\gamma^\alpha$
+
+$$n_\gamma(r)=\sum_\alpha\int\rho_\alpha(r')\omega_\gamma^\alpha(r-r')dr'\tag{1}$$
+
+In practice the weight functions tend to have simple shapes like step functions (i.e. the weighted density is an average over a sphere) or Dirac distributions (i.e. the weighted density is an average over the surface of a sphere).
+
+For Helmholtz energy functionals that can be written in this form, the calculation of the functional derivative can be automated. In general the functional derivative can be written as
+
+$$F_{\rho_\alpha}(r)=\int\frac{\delta f(r')}{\delta\rho_\alpha(r)}dr'=\int\sum_\gamma f_{n_\gamma}(r')\frac{\delta n_\gamma(r')}{\delta\rho_\alpha(r)}dr'$$
+
+with $f_{n_\gamma}$ as abbreviation for the partial derivative $\frac{\partial f}{\partial n_\gamma}$. Using the definition of the weighted densities (1), the expression can be rewritten as
```

Comment:
maybe emphasize "partial"?

`...as abbreviation for the *partial* derivative...`